modules/core/StyleSystem.js

```diff
@@ -590,19 +582,67 @@ export class StyleSystem extends AbstractSystem {
     return Promise.resolve();
   }
 
+  /**
+   * getColorScheme
+   * @return {Object}  Default color scheme object
+   */
+  getColorScheme() {
+    return this.currentColorScheme;
+  }
+
+  /**
+   * getAllColorSchemeas
+   * @return {Object}  All color scheme objects
+   */
+  getAllColorSchemes() {
+    return this.colorSchemes;
+  }
+
+  /**
+   * setColorScheme
+   * Assigns the colorData var to the new scheme, if the selected scheme is not the current scheme
+   * @param  {Object}  scheme - color scheme project
+   */
+  setColorScheme(scheme) {
+    let currentScheme = this.colorSchemes[scheme];
+    if (this.currentColorScheme !== currentScheme) {
+      this.currentColorScheme = currentScheme;
+    }
+  }
+
+  /**
+     * setMode
+     * Assigns Focus mode
+     * @param  {String}  mode - Mode Name
+     */
+  setMode(mode) {
+    this.focusMode = mode;
+  }
+
+  /**
+      * getMode
+      * Retrive Focus mode name
+      * @return  {String}  mode - Mode Name
+      */
+  getMode() {
+    return this.focusMode;
+  }
 
   /**
    * styleMatch
    * @param  {Object}  tags - OSM tags to match to a display style
    * @return {Object}  Styling info for the given tags
    */
+  // eslint-disable-next-line complexity
```

Comment:
Please don't silence the warnings about code complexity.
I think these are useful.